### PR TITLE
demo/chbench: restore original simplicity

### PR DESCRIPTION
**Pull request by benesch**
Restore chbench as a true user-facing demo. It was previously doing
triple duty as a demo, correctness test, and load test, which made the
code inscrutable.

The correctness test bit wasn't used at all, so this commit just ditches
that entirely. I do need to figure out how to restore the load test,
though, before this is mergeable.

### Motivation

   * This PR refactors existing code.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

**Comment by benesch** [comment]
The load test functionality is now restored with a tiny change that allows passing arguments into the `chbench` load generator, which is used by the load tests to configure some peeks, but _not_ by the demo.

I've preserved the removal of peeker because it seems entirely redundant with the peeking ability built into the chBenchmark program. Would anyone care to contest this?

File: demo/chbench/mzworkflows.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+
+def workflow_load_test(w: Workflow):
+    """Run CH-benCHmark with a selected amount of load against Materialize."""
+    w.start_services(services=["prometheus-sql-exporter"])
+    workflow_demo(
+        w,
+        [
+            "--peek-conns=1",
+            "--mz-views=q01,q02,q05,q06,q08,q09,q12,q14,q17,q19",
```
Comment (quodlibetor) [inline demo/chbench/mzworkflows.py:85]: where is `mz-views` defined?

Comment (benesch) [reply demo/chbench/mzworkflows.py:85]: https://github.com/MaterializeInc/materialize/blob/32f3600a714d669fbeb3ed3c808deebf82efb201/demo/chbench/chbench/src/chBenchmark.cc#L377

https://github.com/MaterializeInc/materialize/blob/32f3600a714d669fbeb3ed3c808deebf82efb201/demo/chbench/chbench/src/chBenchmark.cc#L624-L641

https://github.com/MaterializeInc/materialize/blob/main/demo/chbench/chbench/mz-default-mysql.cfg

Comment (quodlibetor) [reply demo/chbench/mzworkflows.py:85]: oh yeah I see.

**Comment by quodlibetor** [comment]
> I've preserved the removal of peeker because it seems entirely redundant with the peeking ability built into the chBenchmark program. Would anyone care to contest this?

This seems fine. Peeker was generally more flexible in terms of being able to dynamically define views and collections of views to run in different thread pools, but I'm pretty sure that I'm the only person who used that so it seems fine. It also reported client-visible query durations to prometheus, but again, most people don't seem to care about that, so we can ditch it.

**Review by quodlibetor — APPROVED** [review]

**Comment by benesch** [comment]
Ok, great! Thanks very much for taking a look.

**Pull request merged**